### DISCUSSION TITLE
Got no premarket data for stock market untill main trading session started

### DISCUSSION THREAD

**Discussion by andreyversh** [ranaroussi/yfinance#1489] [General]
Hello,  please share the code sample to get data for premarket. I use
df = yf.download(tickers='GOOG', period=timeslot, interval='1m', rounding=True, group_by='Ticker', prepost=True),
where timeslot covered premarket time from 4:00 to 09:30 EST, but got all data for premarket only, when main session of stock market is started.

**Comment by ValueRaider** [comment]
@lebedev-a FYI

**Reply by lebedev-a** [reply]
Right, I think @andreyversh is stating the same issue.  Pre-mkt data becomes available only after the market open.

**Comment by lebedev-a** [comment]
Here is what I tried and I know doesn't work:

I've tried pulling data via a cron job  that dumps the raw JSON for a later analysis and can definitely confirm that the following query returns pre-mkt data only after 9:30 am.

`curl -o $path/$filename "https://query1.finance.yahoo.com/v8/finance/chart/$ticker?includePrePost=true&range=2d" `
[ - the rest of the cmd is truncated for readibility]

Run at 9:29am on the 13th
`./parse.py AAPL_2023-07-13_09-29-57.json| grep '\-13 ' |wc -l`
0

Run at 9:30am on the 13th
` ./parse.py AAPL_2023-07-13_09-30-01.json|grep '\-13 0[45678]'|wc -l`
293


[Edits for readability and better output]

**Reply by ValueRaider** [reply]
#1608 might help. No idea what it does because weekend.

**Reply by lebedev-a** [reply]
Very interesting. 
Can't seem to find an explanation of what yahoo feed Id param is and exactly is doing. 
Scheduled a job with  0,28,31 8-9  to grab me some data for  large and small volume ticker.

**Reply by lebedev-a** [reply]
Sadly the same thing
` "https://query1.finance.yahoo.com/v8/finance/chart/$ticker?includePrePost=true&range=2d&useYfid=true" \`
> ./parse.py AAPL_2023-07-17_09-28-01.json| grep '\-17 '|wc -l
0
> ./parse.py AAPL_2023-07-17_09-31-01.json| grep '\-17 0[45678]' |wc -l
284
